aeon/base/_base_collection.py

```
@@ -212,6 +212,33 @@ def _convert_X(self, X):
         X = convert_collection(X, inner_type)
         return X
 
+    def _check_shape(self, X):
+        """Check that the shape of X is consistent with the data seen in fit.
+
+        Parameters
+        ----------
+        X : data structure
+            Must be of type aeon.registry.COLLECTIONS_DATA_TYPES.
+        """
+        if not self.get_tag("capability:unequal_length"):
+            if get_n_timepoints(X) != self.metadata_["n_timepoints"]:
+                raise ValueError(
+                    "X has different length to the data seen in fit but "
+                    "this classifier cannot handle unequal length series."
+                    "length of train set was",
+                    self.metadata_["n_timepoints"],
+                    " length in predict is ",
+                )
+        if self.get_tag("capability:multivariate"):
+            if get_n_channels(X) != self.metadata_["n_channels"]:
```

maybe redundent, but should we also check if same length is on all channels when "capability:unequal_length" is false ?

---

If there are a different number of channels at any point it should throw an error, thats not supported. 

Unequal length data where the unequal tag is false will be picked up in data preprocessing.

Only things we want to check are consistent in both `fit` to `predict` is these two I think.